website/addons/osfstorage/model.py

```diff
@@ -136,139 +102,131 @@ def create_waterbutler_log(self, auth, action, metadata):
         pass
 
 
-class BaseFileObject(StoredObject):
+class OsfStorageFileNode(StoredObject):
     __indices__ = [
         {
             'key_or_list': [
-                ('path', pymongo.ASCENDING),
+                ('name', pymongo.ASCENDING),
```

Comment:
Just a note: once #2181 is merged, you will be able to use the `unique_on` decorator, like so:

```python
@unique_on(['name', 'kind', 'parent', 'node_settings'])
class OsfStorageFileNode(StoredObject):
   # ...
```